src/transformers/tokenization_utils_fast.py

```diff
@@ -618,12 +618,16 @@ def _decode(
         clean_up_tokenization_spaces: bool = None,
         **kwargs,
     ) -> str:
+        # Popping out the allowed unexpected kwargs first
         self._decode_use_source_tokenizer = kwargs.pop("use_source_tokenizer", False)
+        self._spaces_between_special_tokens = kwargs.pop("spaces_between_special_tokens", True)
```

Comment:
Thanks, I think that we should rather add `use_source_tokenizer` as an arg and that's it! (no kwargs)
(Also we should propagate to the slow tokenizers as well: 
```python 
In [11]: from transformers import AutoTokenizer
    ...: 
    ...: tokenizer = AutoTokenizer.from_pretrained("bert-base-cased", use_fast = False)
    ...: 
    ...: string = "I am a string"
    ...: 
    ...: # works
    ...: tokens = tokenizer(string)
    ...: 
    ...: # works
    ...: new_string = tokenizer.decode(tokens["input_ids"])
    ...: 
    ...: # works
    ...: new_string = tokenizer.decode(tokens["input_ids"], invalid_kwargs_argument=True)
    ...: 
    ...: # produces error
    ...: tokens = tokenizer(string, invalid_kwargs_argument=True)
    ...: 
    ...: # produces error
    ...: tokens = tokenizer.encode(string, invalid_kwargs_argument=True)
Keyword arguments {'invalid_kwargs_argument': True} not recognized.
Keyword arguments {'invalid_kwargs_argument': True} not recognized.
```
this won't produce an error for the slow tokenizer.

Reply:
@ArthurZucker One thing that I notice in the slow tokenizer part under tokenization_utils.py is that the kwargs is being propagated to other functions internally hence I am not sure if the same thing can be done there. Please clarify. Thanks